docker-entrypoint.sh

```diff
@@ -1,17 +1,21 @@
 #!/usr/bin/env bash
-set -euo pipefail
+set -eo pipefail
+shopt -s extglob
 
-generate_config() {
-  if [ -z "${POSTGRES_HOST}" ] || [ -z "${POSTGRES_PORT}" ] \
-  || [ -z "${POSTGRES_DB}" ] || [ -z "${POSTGRES_USER}" ] \
-  || [ -z "${POSTGRES_PASSWORD}" ]; then
+warn() {
+  >&2 echo "$@"
+}
+
+generate_xnat_conf() {
+  if [ -z "${POSTGRES_HOST}" ] || [ -z "${POSTGRES_DB}" ] \
+  || [ -z "${POSTGRES_USER}" ] || [ -z "${POSTGRES_PASSWORD}" ]; then
     echo "Error: Database configuration required" >&2
     exit 1
   fi
 
   cat > /data/xnat/home/config/xnat-conf.properties << EOF
 datasource.driver=org.postgresql.Driver
-datasource.url=jdbc:postgresql://${POSTGRES_HOST}:${POSTGRES_PORT}/${POSTGRES_DB}
+datasource.url=jdbc:postgresql://${POSTGRES_HOST}:${POSTGRES_PORT:-5432}/${POSTGRES_DB}
```

Comment:
what is the -5432 here?

Reply:
The `:-` says to bash "If `POSTGRES_PORT` is  not set, set it to whatever is after the `:-`." It's called a parameter expansion.

https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html

Reply:
oh, nice. I just learned something new!